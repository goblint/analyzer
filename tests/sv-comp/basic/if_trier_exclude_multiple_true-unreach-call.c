extern void __VERIFIER_error() __attribute__ ((__noreturn__));
extern int __VERIFIER_nondet_int();
void __VERIFIER_assert(int cond) {
    if (!(cond)) {
        ERROR: __VERIFIER_error();
    }
    return;
}

int main()
{
    while (1)
    {
        int x = __VERIFIER_nondet_int();
        if (x == 0)
        {
            __VERIFIER_assert(x == 0);
        }
        else if (x == 2)
        {
            __VERIFIER_assert(x == 2);
        }
        else
        {
            __VERIFIER_assert(x != 0 && x != 2);
        }
    }
    return 0;
}