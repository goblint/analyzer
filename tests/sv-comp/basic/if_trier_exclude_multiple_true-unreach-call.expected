
                                                                                                ┌·······················································································┐
                                                                                                :                                                                                       :
                                                                                                :                                                                                       :
  ┌─────────────────────────────────────────────────────────────────────────────────────────────┼───────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────┐
  │                                                                                             :                                                                                       :                                                                                                  │
  │                                                                                             :                                           ┌────────────────────────────────────────┐  :                                                                                                  │
  │                                                                                             :                                           │                   _                    │  :                                                                                                  │
  │                                                                                             :                                           └────────────────────────────────────────┘  :                                                                                                  │
  │                                                                                             :                                             │                                         :                                                                                                  │
  │                               ┌·····························································┘                                             │ Entry main                              : Inlined Proc '__VERIFIER_assert(x == 0)'                                                         │
  │                               :                                                                                                           ▼                                         ▼                                                                                                  │
  │                               :                                                            InlineReturn                                 ┌──────────────────────────────────────────────────────────────────────────────────────────────┐  Inlined Proc '__VERIFIER_assert(x == 2)'   ┌─────────────────────┐
  │                          ┌────┼───────────────────────────────────────────────────────────────────────────────────────────────────────▶ │                                                                                              │ ◀·········································· │          _          │
  │                          │    :                                                                                                         │                                                                                              │                                             └─────────────────────┘
  │                          │    :                                                            InlineReturn                                 │                                                                                              │                                               ▲
  │                          │    :                                 ┌─────────────────────────────────────────────────────────────────────▶ │                                              _                                               │                                               │ Test (x == 2,true)
  │                          │    :                                 │                                                                       │                                                                                              │                                               │
  │                          │    :                                 │                                                                       │                                                                                              │                                               │
  │                          │    :                                 │                           ┌·········································▶ │                                                                                              │ ◀┐                                            │
  │                          │    :                                 │                           :                                           └──────────────────────────────────────────────────────────────────────────────────────────────┘  │                                            │
  │                          │    :                                 │                           :                                             │                                                                                               │                                            │
  │                          │    :                                 │                           :                                             │ Test (1,true)                                                                                 │ InlineReturn                               │
  │                          │    :                                 │                           :                                             ▼                                                                                               │                                            │
  │                          │    :                                 │                           :                                           ┌────────────────────────────────────────┐                                                        │                                            │
  │                          │    :                                 │                           :                                           │                   _                    │ ·┐                                                ┌────┼────────────────────────────────────────────┘
  │                          │    :                                 │                           :                                           └────────────────────────────────────────┘  :                                                │    │
  │                          │    :                                 │                           :                                             │                                         :                                                │    │
  │                          │    :                                 │                           :                                             │ InlineEntry '()'                        :                                                │    │
  │                          │    :                                 │                           :                                             ▼                                         :                                                │    │
  │                          │    :                                 │                           :                                           ┌────────────────────────────────────────┐  :                                                │    │
  │                          │    :                                 │                           :                                           │                   _                    │  :                                                │    │
  │                          │    :                                 │                           :                                           └────────────────────────────────────────┘  :                                                │    │
  │ InlineEntry '(x == 2)'   │    :                                 │                           :                                             │                                         :                                                │    │
  │                          │    :                                 │                           :                                             │ Entry __VERIFIER_nondet_int             :                                                │    │
  │                          │    :                                 │                           :                                             ▼                                         :                                                │    │
  │                          │    :                                 │                           :                                           ┌────────────────────────────────────────┐  :                                                │    │
  │                          │    :                                 │                           : Inlined Proc '__VERIFIER_assert(tmp___0)' │                   _                    │  : Inlined Proc 'tmp = __VERIFIER_nondet_int()'   │    │
  │                          │    :                                 │                           :                                           └────────────────────────────────────────┘  :                                                │    │
  │                          │    :                                 │                           :                                             │                                         :                                                │    │
  │                          │    :                                 │                           :                                             │ Ret (Some val, __VERIFIER_nondet_int)   :                                                │    │
  │                          │    :                                 │                           :                                             ▼                                         :                                                │    │
  │                          │    :                                 │                           :                                           ┌────────────────────────────────────────┐  :                                                │    │
  │                          │    :                                 │                           :                                           │                   _                    │  :                                                │    │
  │                          │    :                                 │                           :                                           └────────────────────────────────────────┘  :                                                │    │
  │                          │    :                                 │                           :                                             │                                         :                                                │    │
  │                          │    :                                 │                           :                                             │ InlineReturn 'tmp'                      :                                                │    │
  │                          │    :                                 │                           :                                             ▼                                         :                                                │    │
  │                          │    :                                 │                           :                                           ┌────────────────────────────────────────┐  :                                                │    │
  │                          │    :                                 │                           :                                           │                   _                    │ ◀┘                                                │    │
  │                          │    :                                 │                           :                                           └────────────────────────────────────────┘                                                   │    │
  │                          │    :                                 │                           :                                             │                                                                                          │    │
  │                          │    :                                 │                           :                                             │ Assign 'x = tmp'                                                                         │    │
  │                          │    :                                 │                           :                                             ▼                                                                                          │    │
  │                          │  ┌────────────────────────────────┐  │   Test (x == 0,true)      :                                           ┌────────────────────────────────────────┐                                                   │    │
  │                          │  │               _                │ ◀┼───────────────────────────┼────────────────────────────────────────── │                   _                    │                                                   │    │
  │                          │  └────────────────────────────────┘  │                           :                                           └────────────────────────────────────────┘                                                   │    │
  │                          │    │                                 │                           :                                             │                                                                                          │    │
  │                          │    │ InlineEntry '(x == 0)'          │                           :                                             │ Test (x == 0,false)                                                                      │    │
  │                          │    ▼                                 │                           :                                             ▼                                                                                          │    │
  │                          │  ┌────────────────────────────────┐  │                           :                                           ┌────────────────────────────────────────┐                                                   │    │
  └──────────────────────────┼▶ │               _                │  │                           :                                           │                   _                    │ ──────────────────────────────────────────────────┘    │
                             │  └────────────────────────────────┘  │                           :                                           └────────────────────────────────────────┘                                                        │
                             │    │                                 │                           :                                             │                                                                                               │
                             │    │ Entry __VERIFIER_assert         │                           :                                             │ Test (x == 2,false)                                                                           │
                             │    ▼                                 │                           :                                             ▼                                                                                               │
                             │  ┌────────────────────────────────┐  │                           :                                           ┌────────────────────────────────────────┐                                                        │
                             │  │               _                │  │                           :                                           │                   _                    │                                                        │
                             │  └────────────────────────────────┘  │                           :                                           └────────────────────────────────────────┘                                                        │
                             │    │                                 │                           :                                             │                                                                                               │
                             │    │ Test (! cond,false)             │                           :                                             │ Assign 'tmp___0 = x != 0 && x != 2'                                                           │
                             │    ▼                                 │                           :                                             ▼                                                                                               │
                             │  ┌────────────────────────────────┐  │                           :                                           ┌────────────────────────────────────────┐                                                        │
                             │  │               _                │  │                           └·········································· │                   _                    │                                                        │
                             │  └────────────────────────────────┘  │                                                                       └────────────────────────────────────────┘                                                        │
                             │    │                                 │                                                                         │                                                                                               │
                             │    │ Ret (None, __VERIFIER_assert)   │                                                                         │ InlineEntry '(tmp___0)'                                                                       │
                             │    ▼                                 │                                                                         ▼                                                                                               │
                             │  ┌────────────────────────────────┐  │                                                                       ┌────────────────────────────────────────┐                                                        │
                             └─ │               _                │ ─┘                                                                       │                   _                    │                                                        │
                                └────────────────────────────────┘                                                                          └────────────────────────────────────────┘                                                        │
                                                                                                                                              │                                                                                               │
                                                                                                                                              │ Entry __VERIFIER_assert                                                                       │
                                                                                                                                              ▼                                                                                               │
                                                                                                                                            ┌────────────────────────────────────────┐                                                        │
                                                                                                                                            │                   _                    │                                                        │
                                                                                                                                            └────────────────────────────────────────┘                                                        │
                                                                                                                                              │                                                                                               │
                                                                                                                                              │ Test (! cond,false)                                                                           │
                                                                                                                                              ▼                                                                                               │
                                                                                                                                            ┌────────────────────────────────────────┐                                                        │
                                                                                                                                            │                   _                    │                                                        │
                                                                                                                                            └────────────────────────────────────────┘                                                        │
                                                                                                                                              │                                                                                               │
                                                                                                                                              │ Ret (None, __VERIFIER_assert)                                                                 │
                                                                                                                                              ▼                                                                                               │
                                                                                                                                            ┌────────────────────────────────────────┐                                                        │
                                                                                                                                            │                   _                    │ ───────────────────────────────────────────────────────┘
                                                                                                                                            └────────────────────────────────────────┘
